core/templates/domain/story/editable-story-backend-api.service.ts

```diff
@@ -16,203 +16,252 @@
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+
+interface FetchStoryBackendResponse {
+  'story': StoryBackendDict;
+  'topic_name': string;
+  'story_is_published': boolean;
+  'skill_summaries': SkillSummaryBackendDict[];
+  'topic_url_fragment': string;
+  'classroom_url_fragment': string;
+}
+
+interface FetchStoryResponse {
+  story: StoryBackendDict;
+  topicName: string;
+  storyIsPublished: boolean;
+  skillSummaries: SkillSummaryBackendDict[];
+  topicUrlFragment: string;
+  classroomUrlFragment: string;
+}
+
+interface UpdateStoryBackendResponse {
+  story: StoryBackendDict;
+}
+
+interface StoryUrlFragmentExists {
+  // eslint-disable-next-line
+  story_url_fragment_exists: StoryUrlFragmentExists;
+}
+
+interface ValidationErrorMessages {
+  // eslint-disable-next-line
+  validation_error_messages: ValidationErrorMessages;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class EditableStoryBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolationService : UrlInterpolationService) {}
+
+  private _fetchStory(
+      storyId: string,
+      successCallback: (value: FetchStoryResponse) => void,
+      errorCallback: (reason: string) => void): void {
+    const editableStoryDataUrl = this.urlInterpolationService.interpolateUrl(
+      StoryDomainConstants.EDITABLE_STORY_DATA_URL_TEMPLATE, {
+        story_id: storyId
       });
-    };
 
-    var _updateStory = function(
-        storyId, storyVersion, commitMessage, changeList,
-        successCallback, errorCallback) {
-      var editableStoryDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_STORY_DATA_URL_TEMPLATE, {
-          story_id: storyId
+    this.http.get<FetchStoryBackendResponse>(
+      editableStoryDataUrl).toPromise().then(response => {
+      if (successCallback) {
+        successCallback({
+          story: response.story,
+          topicName: response.topic_name,
+          storyIsPublished: response.story_is_published,
+          skillSummaries: response.skill_summaries,
+          topicUrlFragment: response.topic_url_fragment,
+          classroomUrlFragment: response.classroom_url_fragment
         });
+      }
+    }, errorResponse => {
+      if (errorCallback) {
+        errorCallback(errorResponse.error.error);
+      }
+    });
+  }
 
-      var putData = {
-        version: storyVersion,
-        commit_message: commitMessage,
-        change_dicts: changeList
-      };
-      $http.put(editableStoryDataUrl, putData).then(function(response) {
-        // The returned data is an updated story dict.
-        var story = angular.copy(response.data.story);
-
-        if (successCallback) {
-          successCallback(story);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+  private _updateStory(
+      storyId: string, storyVersion: string,
```

Comment:
Is `storyVersion` a `string`?

Reply:
In the spec file it is of type string so

Reply:
@ReshuKumari no the spec file specifies the version as a number

Reply:
```
sampleDataResults = {
      story: {
        id: 'storyId',
        title: 'Story title',
        description: 'Story description',
        notes: 'Notes',
        version: 1,
        story_contents: {
          initial_node_id: 'node_1',
          nodes: [{
            id: 'node_1',
            prerequisite_skill_ids: [],
            acquired_skill_ids: [],
            destination_node_ids: [],
            outline: 'Outline',
            exploration_id: null,
            outline_is_finalized: false
          }],
          next_node_id: 'node_3'
        },
        language_code: 'en'
      },
      topic_name: 'Topic Name',
      story_is_published: true,
      skill_summaries: [{
        id: 'skill_1',
        description: 'Skill Description'
      }],
      topic_url_fragment: 'topic-frag',
      classroom_url_fragment: 'math'
    };
```

Reply:
@darksun27 yes, corrected.

Reply:
Thanks, @ReshuKumari! Can you please push those changes?